server/py/services/api/crud/model_monitoring/deployment.py

```diff
@@ -741,16 +750,203 @@ def _create_tsdb_tables(
+
+        :return: Returns a tuple containing a list of FunctionSummary objects and the base period of the controller
+                 function. If `include_infra` is False, the list will be empty.
+
+        """
+        function_summaries_list = []
+        base_period = 0
+        if include_infra:
+            infra_mm_functions = self.list_model_monitoring_functions(
+                format_=mlrun.common.formatters.FunctionFormat.full, infra_only=True
+            )
+
+            if not infra_mm_functions:
+                logger.info("No model monitoring infrastructure functions found")
+            for function in infra_mm_functions:
+                function_summary = (
+                    mlrun.common.schemas.model_monitoring.FunctionSummary.from_dict(
+                        function, func_type="infra"
+                    )
+                )
+                function_summaries_list.append(function_summary)
+                if (
+                    function["metadata"]["name"]
+                    == mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER
+                ):
+                    base_period = self._get_base_period(controller_func=function)
+        else:
+            # getting the base period from the controller function
+            try:
+                controller_func = services.api.crud.Functions().get_function(
+                    db_session=self.db_session,
+                    name=mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER,
+                    project=self.project,
+                )
+                base_period = self._get_base_period(controller_func=controller_func)
+            except mlrun.errors.MLRunNotFoundError:
+                logger.info(
+                    "Model monitoring controller function not found, "
+                    "probably you need to re-enable model monitoring",
+                    project=self.project,
+                )
+        return function_summaries_list, base_period
+
+    def _enrich_function_summary_with_applications(
+        self,
+        functino_summaries_list: list,
+        base_period: typing.Optional[float] = None,
+        start: typing.Optional[datetime] = None,
+        end: typing.Optional[datetime] = None,
+        names: typing.Optional[list[str]] = None,
+        labels: typing.Optional[list[str]] = None,
+        include_stats: bool = True,
+    ):
+        """
+        Enrich the function summaries list with the model monitoring applications.
+        """
+        mm_functions = self.list_model_monitoring_functions(
+            labels=labels, format_=mlrun.common.formatters.FunctionFormat.minimal
         )
+        if names:
+            mm_functions = [
+                fn for fn in mm_functions if fn["metadata"]["name"] in names
+            ]
+        if not mm_functions:
+            logger.info("No model monitoring applications found")
+            return functino_summaries_list
```

Comment:
It makes more logical sense to switch the ordering here.
```suggestion
        if not mm_functions:
            logger.info("No model monitoring applications found")
            return function_summaries_list

        if names:
            mm_functions = [
                fn for fn in mm_functions if fn["metadata"]["name"] in names
            ]
```